shared/js/background/declarative-net-request.js

```diff
@@ -429,7 +437,36 @@ export async function updateUserDenylist () {
     await updateExtensionConfigRules()
 }
 
+/**
+ * Ensure that the allowing rule for ServiceWorker initiated requests is
+ * enabled. Since the rule needs to be restricted to matching requests not
+ * associated with a tab (tabId of -1) and so must be a session rule. Session
+ * rules don't persist past a browsing session, so must be re-added.
+ * Note: Only exported for use by unit tests, do not call manually.
+ * @return {Promise}
+ */
+export async function ensureServiceWorkerInitiatedRequestException () {
+    const removeRuleIds = [SERVICE_WORKER_INITIATED_ALLOWING_RULE_ID]
+    const addRules = [generateDNRRule({
```

Comment:
Did you want to restrict this to `xmlhttprequest` or do we need to allow all types?

Reply:
I'd prefer to do all types, since that matches the MV2 logic. (I agree that might include other requests, but I figure the main thing for now is to be consistent between MV2 and MV3.)